src/mpol/images.py

```diff
@@ -279,6 +280,120 @@ def to_FITS(self, fname="cube.fits", overwrite=False, header_kwargs=None):
 
         hdul.close()
 
+class PBCorrectedCube(nn.Module):
+    r"""
+    A ImageCube that has been corrected for the expected primary beam. Currently only implemented for ALMA.
+    
+     Args:
+        cell_size (float): the width of a pixel [arcseconds]
+        npix (int): the number of pixels per image side
+        coords (GridCoords): an object already instantiated from the GridCoords class. If providing this, cannot provide ``cell_size`` or ``npix``.
+        nchan (int): the number of channels in the image
+    """
+    def __init__(
+        self,
+        cell_size=None,
+        npix=None,
+        coords=None,
+        chan_freqs=None,
+        telescope="ALMA_12"
+    ):
+        nchan = len(np.at_least_1d(chan_freqs))
+        super().__init__()
+        _setup_coords(self, cell_size, npix, coords, nchan)
+        
+        self.telescope_to_radius = {"ALMA_12": 6.} # in meters
+        self.telescope_to_blocked_radius = {"ALMA_12": 0.375}
+        
+        assert (telescope in self.telescope_to_radius) and (telescope in self.telescope_to_blocked_radius)
+        
+        self.pb_mask = self.obscured_airy_disk(chan_freqs, telescope)
```

Comment:
You might want to add a comment here noting that pb_mask is actually in packed cube format. I need to be a bit better about specifying these throughout the ImageCube too.